crates/holochain_conductor_api/src/app_interface.rs

```diff
@@ -33,6 +33,19 @@ pub enum AppRequest {
     /// [`AppResponse::ZomeCalled`]
     CallZome(Box<ZomeCall>),
 
+    /// Revoke an agent key for an app.
+    ///
+    /// When an agent key is revoked, it becomes invalid and can no longer be used to author
+    /// actions for that app. The source chain of all cells of the app are made read-only.
+    /// No further actions can be added to the chains by anyone. They can still be queried.
+    ///
+    /// Cloning a cell of this app will fail, as the new clone cell could not be written to.
+    ///
+    /// # Returns
+    ///
+    /// [`AppResponse::AgentKeyRevoked`]
+    RevokeAgentKey(Box<RevokeAgentKeyPayload>),
```

Comment:
Hmm, do we really want the app to be able to do this at will? Maybe so, but I thought maybe it would be an admin thing?

Reply:
When we spoke about this a week ago, you thought it should be the app interface. I don't have an opinion about this, as I don't know where and how key revocation will be used.

Reply:
I don't know where it will be used either. Sorry if I said it should be app interface. Currently I'm not seeing a valid reason why an app UI should ever be allowed to revoke a key.

Reply:
Why shouldn't the app be able to do this? The key will only be used with the app if I'm understanding correctly, so the scope is already at the app level. 

So the worst thing a malicious app could do is prevent you from using that app. It can't affect other apps.

Reply:
It's more than the danger of preventing you from using the app, it's the danger of bricking your source chain forever so you can't write to it anymore.